api/controllers/v0.3/ILSHelper.js

```diff
@@ -0,0 +1,390 @@
+  }
+
+  // Format a patron's address as ILS expects it.
+  formatAddress(address) {
+    return address.toUpperCase.gsub(/[\n\r]+/, "$");
+  }
+
+  // Internal: Format card fields for ILS patron creation/updates.
+  formattedPatronData(patron) {
+    let email;
+    let birthdate;
+    let workAddress;
+    if (patron.email) {
+      email = patron.email.gsub(/\s+/, "").toUpperCase;
+    }
+    if (patron.birthdate) {
+      // birthdate = patron.birthdate.strftime("%Y%m%d000000");
+      birthdate = patron.birthdate;
+    }
+
+    let address = this.formatAddress(patron.address);
+    if (patron.worksInCity) {
+      workAddress = this.formatAddress(patron.workAddress);
+    }
+
+    let fullName = patron.name.toUpperCase();
+    if (!fullName.includes(",")) {
+      // Existing TODO: Replace this code with a call to a dedicated human name
+      // parsing library.
+
+      // Simplistically assume person's name has two segments: the first
+      // and last name, with no middle names or honorifics.
+      // (e.g. "Dorothy Vaughn" "Katherine Johnson" "Mary Jackson")
+      let [firstName, lastName] = fullName.split(" ", 2);
+      firstName = firstName.trim();
+      lastName = lastName.trim();
+
+      if (firstName !== "" || lastName !== "") {
+        fullName = `//${lastName}, //${firstName}`;
+      } else {
+        // Our assumption of at least two names was wrong, and we got
+        // something like "Dorothy" or "Katherine " or " Jackson".
+        // We'll take what we can get.
+        fullName = firstName || lastName;
+      }
+    }
+
+    let fields = {
+      name: fullName,
+      address: address,
+      username: patron.username,
+      pin: patron.pin,
+      ptype: patron.ptype,
+      workAddress: workAddress,
+      ecommunications_pref: patron.ecommunicationsPref,
+    };
+
+    if (patron.barcode) {
+      fields["barcode"] = patron.barcode;
+    }
+    if (email) {
+      fields["email"] = email;
+    }
+    if (birthdate) {
+      fields["birthdate"] = birthdate;
+    }
+
+    return fields;
+  }
+
+  // Determine whether a patron should have a permanent card.
+  longExpiration(patron) {
+    // False if a patron's existing work address isn't commercial
+    if (patron.workAddress && patron.workAddress.isResidential) {
+      return false;
+    }
+
+    // False if patron provides a home address that is not residential
+    // False if patron does not have a recognized name
+    // False if patron policy is not the default (:simplye)
+    return (
+      patron.address.isResidential &&
+      patron.hasValidName &&
+      patron.policy.isDefault
+    );
+  }
+
+  patronFields(patron) {
+    let fieldData = this.formattedPatronData(patron);
+
+    let fields = [
+      fieldData["name"],
+      fieldData["address"],
+      fieldData["ptype"],
+      fieldData["username"],
+      fieldData["pin"],
+      fieldData["ecommunications_pref"],
+      // notice_pref_as_field,
+      // home_lib_as_field,
```

Comment:
Were these supposed to be uncommented/removed?

Reply:
I have a better idea of how this works and much of this can be removed. I didn't understand how those values were created in ruby so just commented it out for now.